graylog2-server/src/main/java/org/graylog2/streams/matchers/ExactMatcher.java

```diff
@@ -26,17 +26,13 @@
 
 	@Override
 	public boolean match(Message msg, StreamRule rule) {
-        if (!msg.hasField(rule.getField())) {
-            return false;
+        if (msg.getField(rule.getField()) == null) {
+            return rule.getInverted();
         }
 
-		Object value = msg.getField(rule.getField());
-		
-		if (value == null) {
-			return false;
-		}
-		
-		return rule.getInverted() ^ value.toString().trim().equals(rule.getValue());
+		final String value = msg.getField(rule.getField()).toString();
+
+		return rule.getInverted() ^ value.trim().equals(rule.getValue());
```

Comment:
✅ I interpreted the intention of that expression completely wrong. Everything's fine.

![Everything's fine](https://cloud.githubusercontent.com/assets/43951/14984253/04a8179c-1142-11e6-870d-2781b2ddb820.png)